pyam/utils.py

```diff
@@ -194,18 +235,18 @@ def format_data(df):
     # cast value columns to numeric, drop NaN's, sort data
     df['value'] = df['value'].astype('float64')
     df.dropna(inplace=True)
-    df.sort_values(META_IDX + ['variable', time_col, 'region'] + extra_cols,
-                   inplace=True)
 
-    return df, time_col, extra_cols
+    # check for duplicates and return sorted data
+    idx_cols = IAMC_IDX + [time_col] + extra_cols
+    if any(df[idx_cols].duplicated()):
+        raise ValueError('duplicate rows in `data`!')
 
+    return sort_data(df, idx_cols), time_col, extra_cols
 
-def style_df(df, style='heatmap'):
-    if style == 'highlight_not_max':
-        return df.style.apply(lambda s: ['' if v else 'background-color: yellow' for v in s == s.max()])
-    if style == 'heatmap':
-        cm = sns.light_palette("green", as_cmap=True)
-        return df.style.background_gradient(cmap=cm)
+
+def sort_data(data, cols):
+    """Sort `data` rows and order columns"""
+    return data.sort_values(cols)[cols + ['value']].reset_index(drop=True)
```

Comment:
what happens if "value" is already in `cols`? not sure if this is a corner case, but could do `list(set(cols + ['value']))`

Reply:
Usig `list(set(cols + ['value']))` would defeat the purpose of having a consistent ordering of columns, which makes the unit tests a lot more straightforward by using `pd.testing.assert_frame_equal()`.

I tried migrating that function to `core.py`, but then this function can't be used in the unit tests, which is opening up another can of worms.